composites/Plugin/SnippetPreview/components/SnippetPreview.js

```diff
@@ -1,17 +1,625 @@
+			} );
+		}
+	}
+
+	/**
+	 * Removes the last word of a sentence.
+	 *
+	 * @param {string} sentence The sentence to drop a word of.
+	 * @returns {string} The new sentence.
+	 */
+	dropLastWord( sentence ) {
+		let titleParts = sentence.split( " " );
+		titleParts.pop();
+
+		return titleParts.join( " " );
+	}
+
+	/**
+	 * Returns the title for rendering.
+	 *
+	 * @returns {string} The title to render.
+	 */
+	getTitle() {
+		if ( this.props.title !== this.state.title ) {
+			return this.state.title + " ...";
+		}
+
+		return this.props.title;
+	}
+
+	/**
+	 * Renders the description for display
+	 *
+	 * @returns {ReactElement} The rendered description.
+	 */
+	renderDescription() {
+		if ( this.props.mode === MOBILE && this.props.description !== this.state.description ) {
+			return this.state.description + " ...";
+		}
+
+		if ( this.props.mode === DESKTOP ) {
+			return truncate( this.props.description, {
+				length: DESCRIPTION_LIMIT,
+				omission: "",
+			} );
+		}
+
+		return this.props.description;
+	}
+
+	/**
+	 * Renders the date if set.
+	 *
+	 * @returns {?ReactElement} The rendered date.
+	 */
+	renderDate() {
+		return this.props.date === ""
+			? null
+			: <DatePreview>{ this.props.date } - </DatePreview>;
+	}
+
+	/**
+	 * Adds carret styles tot the base component if relevant prop is active.
+	 *
+	 * @param {string} fieldName The field to add caret styles to.
+	 * @param {ReactComponent} BaseComponent The base component for the field.
+	 *
+	 * @returns {ReactComponent} The component with caret styles added.
+	 */
+	addCaretStyles( fieldName, BaseComponent ) {
+		const {
+			mode,
+			hoveredField,
+			activeField,
+		} = this.props;
+
+		if ( activeField === fieldName ) {
+			return addCaretStyle( BaseComponent, colorCaret, mode );
+		}
+
+		if ( hoveredField === fieldName ) {
+			return addCaretStyle( BaseComponent, colorCaretHover, mode );
+		}
+
+		return BaseComponent;
+	}
+
+	/**
+	 * Returns the breadcrumbs string to be rendered.
+	 *
+	 * @returns {string} The breadcrumbs.
+	 */
+	getBreadcrumbs() {
+		const { url, breadcrumbs } = this.props;
+		const { protocol, hostname, pathname } = parse( url );
+
+		const hostPart = protocol === "https:" ? protocol + "//" + hostname : hostname;
+
+		const urlParts = breadcrumbs || pathname.split( "/" );
+
+		return [ hostPart, ...urlParts ].filter( part => !! part ).join( " › " );
+	}
+
+	/**
+	 * Renders the URL for display in the snippet preview.
+	 *
+	 * @returns {ReactElement} The rendered URL.
+	 */
+	renderUrl() {
+		const {
+			url,
+			onClick,
+			onMouseOver,
+			onMouseLeave,
+			locale,
+			keyword,
+		} = this.props;
+
+		let urlContent;
+
+		if ( this.props.mode === MOBILE ) {
+			urlContent = this.getBreadcrumbs();
+		} else {
+			urlContent = highlightKeyword( locale, keyword, url );
+		}
+
+		const Url = this.addCaretStyles( "url", BaseUrl );
+
+		return <Url onClick={ onClick.bind( null, "url" ) }
+		            onMouseOver={ partial( onMouseOver, "url" ) }
```

Comment:
Overly pedantic:
There is a combination of tabs and dots here

Reply:
That is intentional. [It is known as smart tabs](https://www.emacswiki.org/emacs/SmartTabs). Because we use spaces for the alignment the attributes will stay aligned even if I choose 7 as the display width for tabs.